### ISSUE TITLE
Plugin registration crashes with NPE

### ISSUE THREAD

**Issue by sergey-triputsco** [amplitude/Amplitude-Flutter#25]
## Expected Behavior
Plugin registers without crashes when app runs in background e.g. Service handle some platform Intent and starts headless FlutterEngine that executes 

## Current Behavior
Amplitude plugin registration crashes:
`     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.BinaryMessenger.setMessageHandler(java.lang.String, io.flutter.plugin.common.BinaryMessenger$BinaryMessageHandler)' on a null object reference
        at io.flutter.plugin.common.MethodChannel.setMethodCallHandler(MethodChannel.java:119)
        at com.amplitude.amplitude_flutter.AmplitudeFlutterPlugin$Companion.registerWith(AmplitudeFlutterPlugin.kt:22)
        at com.amplitude.amplitude_flutter.AmplitudeFlutterPlugin.registerWith(Unknown Source:2)`

## Possible Solution
Migrate the plugin to flutter android v2 embedding.

Don't use `PluginRegistry.messenger()`. According to the docs `This registrar is for Flutter's v1 embedding.`

## Steps to Reproduce
1. Application uses Android v2 plugin embedding.
```    
<application
        android:name=".Application"
        android:icon="@mipmap/ic_launcher"
        ...>

        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
```
2. Execute `GeneratedPluginRegistrant.registerWith(engine)` when an app runs in background e.g. a Service instantiates a FlutterEngine and registers plugins.

Example:
```
package example

import android.app.IntentService
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugins.GeneratedPluginRegistrant
import io.flutter.view.FlutterCallbackInformation
import io.flutter.view.FlutterMain

class ExampleService : IntentService("Example") {
    private var backgroundChannel: MethodChannel? = null

    private var backgroundFlutterEngine: FlutterEngine? = null

    override fun onCreate() {
        super.onCreate()

        if (backgroundFlutterEngine == null) {

            val callbackHandle = retrieveCallbackHandler(....)
            val callbackInfo = try {
                FlutterCallbackInformation.lookupCallbackInformation(callbackHandle)
            } catch (e: IllegalStateException) {
                null
            }

            if (callbackInfo != null) {
                // Create a FlutterEngine which will process the events.
                backgroundFlutterEngine = FlutterEngine(this.applicationContext)

                GeneratedPluginRegistrant.registerWith((backgroundFlutterEngine!!)

                val args = DartExecutor.DartCallback(this.applicationContext.assets, FlutterMain.findAppBundlePath(this.applicationContext)!!, callbackInfo)
                // Start running callback dispatcher code in our background FlutterEngine instance.
                backgroundFlutterEngine!!.dartExecutor.executeDartCallback(args)

                backgroundChannel = MethodChannel(backgroundFlutterEngine!!.dartExecutor.binaryMessenger, "com.example.callback")
            }
        }

    }

    override fun onDestroy() {
        super.onDestroy()
        backgroundFlutterEngine?.destroy()
    }
}
```

## Environment
- SDK Version: 
- Flutter Version: 1.22.2
- Device: any

**Comment by sergey-triputsco** [comment]
Bug fix #28

**Comment by haoliu-amp** [comment]
Thanks! @sergey-triputsco
